Backport: Use -force_load instead of -all_load on macOS and drop the obsolete -noall_load linker option

https://github.com/DiligentGraphics/DiligentCore/commit/63c2ccfdfe5c4944a37b8ab70c13d81cecd50e61

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -329,8 +329,10 @@
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR
     CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     if(PLATFORM_APPLE)
-        set(WHOLE_ARCHIVE_FLAG "-Wl,-all_load" CACHE INTERNAL "all_load flag")
-        set(NO_WHOLE_ARCHIVE_FLAG "-Wl,-noall_load" CACHE INTERNAL "noall_load flag")
+        # Looks like force_load is preferrable over all_load
+        set(WHOLE_ARCHIVE_FLAG "-Wl,-force_load" CACHE INTERNAL "all_load flag")
+        # Option -noall_load is obsolete and there is no -noforce_load
+        set(NO_WHOLE_ARCHIVE_FLAG "" CACHE INTERNAL "noall_load flag")
     else()
         set(WHOLE_ARCHIVE_FLAG "-Wl,--whole-archive" CACHE INTERNAL "whole-archive flag")
         set(NO_WHOLE_ARCHIVE_FLAG "-Wl,--no-whole-archive" CACHE INTERNAL "no-whole-archive flag")
